src/features/device-types/storage/s3.ts

```diff
@@ -128,6 +139,15 @@ export async function fileExists(s3Path: string): Promise<boolean> {
 		await getFileInfo(s3Path);
 		return true;
 	} catch (err) {
+		// catch errors for private device types when running unauthenticated
+		if (s3Client instanceof UnauthenticatedS3Facade) {
+			if ([401, 403].includes(err.statusCode)) {
+				console.warn(
+					`${err.code} (${err.statusCode}): ${s3Path} belongs to a private device type or has incorrect permissions`,
+				);
+				return false;
+			}
+		}
```

Comment:
This is mostly duplicate code with the other log further up in the file, maybe we could consolidate this into a shared function that only logs the warning. Mayne something like this:
```typescript
function logUnauthenticatedWarning(s3Client, s3Path, err): void {
    if (s3Client instanceof UnauthenticatedS3Facade && [401, 403].includes(err.statusCode)) {
        console.warn(
            `${err.code} (${err.statusCode}): ${s3Path} belongs to a private device type or has incorrect permissions`,
        );
    }
}
```

Then we could simplify the `return false` logic by only doing that in the `if` block below with:
```typescript
if ([401, 403, 404].includes(err.statusCode)) {
    return false;
}
```

What do you think?